src/test/java/org/motechproject/prevac/service/VisitRescheduleServiceTest.java

```diff
@@ -0,0 +1,391 @@
+        Boolean ignoreLimitation = true;
+        expectedVisitRescheduleDto.setStartTime(new Time(9, 0));
+        expectedVisitRescheduleDto.setPlannedDate(new LocalDate(2217, 1, 2));
+        expectedVisitRescheduleDto.setIgnoreDateLimitation(ignoreLimitation);
+        expectedVisitRescheduleDto.setVisitId(1L);
+
+        when(visitDataService.findById(1L)).thenReturn(visit);
+        when(visitDataService.update(visit)).thenReturn(visit);
+
+        VisitRescheduleDto actualVisitRescheduleDto = visitRescheduleService.saveVisitReschedule(expectedVisitRescheduleDto, ignoreLimitation);
+
+        // Returned dto should not have earliest and latest return dates
+        expectedVisitRescheduleDto.setEarliestDate(null);
+        expectedVisitRescheduleDto.setLatestDate(null);
+        assertVisitRescheduleDto(expectedVisitRescheduleDto, actualVisitRescheduleDto);
+
+        verify(subjectDataService).update(any(Subject.class));
+        verify(visitDataService).update(any(Visit.class));
+        verify(visitDataService, never()).findByClinicIdVisitPlannedDateAndType(anyLong(), any(LocalDate.class), any(VisitType.class));
+    }
+
+    private VisitScheduleOffset createVisitScheduleOffset(VisitType visitType, Integer timeOffset, Integer earliestDateOffset, Integer latestDateOffset) {
+        VisitScheduleOffset visitScheduleOffset = new VisitScheduleOffset();
+        visitScheduleOffset.setVisitType(visitType);
+        visitScheduleOffset.setTimeOffset(timeOffset);
+        visitScheduleOffset.setEarliestDateOffset(earliestDateOffset);
+        visitScheduleOffset.setLatestDateOffset(latestDateOffset);
+        return visitScheduleOffset;
+    }
+
+    private Visit createVisit(Long id, VisitType visitType, LocalDate date, LocalDate projectedDate, Subject subject) {
+        Visit visit = new Visit();
+        visit.setId(id);
+        visit.setType(visitType);
+        visit.setDate(date);
+        visit.setDateProjected(projectedDate);
+        visit.setSubject(subject);
+
+        subject.getVisits().add(visit);
+
+        return visit;
+    }
+
+    private void checkVisitRescheduleDtoList(List<VisitRescheduleDto> expectedDtos, List<VisitRescheduleDto> resultDtos) {
+        for (int i = 0; i < expectedDtos.size(); i++) {
+            assertVisitRescheduleDto(expectedDtos.get(i), resultDtos.get(i));
+        }
+    }
+    
+    private void assertVisitRescheduleDto(VisitRescheduleDto expected, VisitRescheduleDto actual) {
+        assertEquals(expected.getEarliestDate(), actual.getEarliestDate());
+        assertEquals(expected.getLatestDate(), actual.getLatestDate());
+        assertEquals(expected.getParticipantId(), actual.getParticipantId());
+        assertEquals(expected.getIgnoreDateLimitation(), actual.getIgnoreDateLimitation());
+    }
+
+    private Subject createSubject(String subjectId, String name, String phoneNumber, String address, Language language, //NO CHECKSTYLE ParameterNumber
```

Comment:
it's used only one time, I don't think we need those parameters, just set the subject fields here